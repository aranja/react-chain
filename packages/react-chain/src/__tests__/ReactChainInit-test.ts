import middleware from '../ReactChainInit'
import createSession from '../Session'

describe('ReactChainInit', () => {
  it('should be a function', () => {
    expect(typeof middleware).toBe('function')
  })

  it('should add non-configurable properties', () => {
    const props = [
      'htmlProps',
      'bodyProps',
      'window',
      'head',
      'footer',
      'css',
      'js',
    ]
    const session = createSession()
    middleware(session)
    props.forEach(prop => {
      expect(session).toHaveProperty(prop)
      expect(() => delete session[prop]).toThrowErrorMatchingSnapshot()
    })
  })
})
