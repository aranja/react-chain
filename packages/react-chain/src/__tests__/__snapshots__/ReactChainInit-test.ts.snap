// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactChainInit should add non-configurable properties 1`] = `"Cannot delete property 'htmlProps' of #<Object>"`;

exports[`ReactChainInit should add non-configurable properties 2`] = `"Cannot delete property 'bodyProps' of #<Object>"`;

exports[`ReactChainInit should add non-configurable properties 3`] = `"Cannot delete property 'window' of #<Object>"`;

exports[`ReactChainInit should add non-configurable properties 4`] = `"Cannot delete property 'head' of #<Object>"`;

exports[`ReactChainInit should add non-configurable properties 5`] = `"Cannot delete property 'footer' of #<Object>"`;

exports[`ReactChainInit should add non-configurable properties 6`] = `"Cannot delete property 'css' of #<Object>"`;

exports[`ReactChainInit should add non-configurable properties 7`] = `"Cannot delete property 'js' of #<Object>"`;
